src/vinyldns/serdes.py

```
@@ -87,3 +90,26 @@ def to_json_string(o):
     """
     import json
     return json.dumps(to_dict(o))
+
+
+def map_option(v, f):
+    """
+    Applies the function f to the value if it is not None
+    :param v: A value that maybe None
+    :param f: A function that takes a single argument to apply to v
+    :return: The result of applying f to v; None if v is None
+    """
+    if v:
+        return f(v)
+    else:
+        return None
+
+
+def parse_datetime(s):
+    """
+    Parses the iso formatted date from the string provided
+    :param s: An iso date formatted string
+    :return: A datetime
+    """
+    import dateutil.parser
+    return dateutil.parser.parse(s)
```

Might want to handle `ValueError` and `OverflowError` here in the event of an invalid string input (source: https://dateutil.readthedocs.io/en/stable/parser.html).

---

Yea, I definitely did not try very hard on the error cases here.  We are only ever using it where we expect a date, so let the error bubble up is what I want to do here.